lib/ansible/modules/cloud/amazon/aws_kms.py

```diff
@@ -198,11 +204,11 @@ def do_grant(kms, keyarn, role_arn, granttypes, mode='grant', dry_run=True, clea
         if len(changes_needed) and not dry_run:
             policy_json_string = json.dumps(policy)
             kms.put_key_policy(KeyId=keyarn, PolicyName='default', Policy=policy_json_string)
-    except:
-        raise Exception("{}: // {}".format("e", policy_json_string))
+    except Exception as e:
+        raise Exception("{}: // {}".format(e, policy_json_string))
```

Comment:
probably.  It is a tradeoff between information that you have locally that may help with debugging versus having a complete traceback.  So if ```policy_json_string``` has important information that won't be available in the outer try/except yo may want to do this here.  A solution common in our modules is to call module.fail_json() at this level so you can have access to both the traceback and the additional information.  (But then you have to pass the module into the do_grant() function).